lib/bolt/cli.rb

```diff
@@ -485,38 +486,30 @@ def outputter
       @outputter ||= Bolt::Outputter.for_format(@config[:format])
     end
 
-    def load_task_data(name, modulepath)
-      module_name, file_name = name.split('::', 2)
-      file_name ||= 'init'
-
+    def load_task_data(name, args, modulepath)
+      task_type = nil
+      task = nil
       begin
-        env = Puppet::Node::Environment.create('bolt', modulepath)
-        Puppet.override(environments: Puppet::Environments::Static.new(env)) do
-          data = Puppet::InfoService::TaskInformationService.task_data(
-            env.name, module_name, name
-          )
-
-          file = data[:files].find { |f| File.basename(f, '.*') == file_name }
-          if file.nil?
-            raise Bolt::CLIError, "Failed to load task file for '#{name}'"
+        Dir.mktmpdir('bolt') do |dir|
```

Comment:
Much of this may be unnecessary... would be interested in any suggestions @hlindberg. I think for type lookup, we primarily need the modulepath.

Reply:
I think a directory is required, you could have an empty tmp directory in bolt that is always used for this. Later you may for example want to have a default hiera.yaml there.

Reply:
sorry, you can also do this `Puppet::Pal.in_tmp_environment` and just give a modulepath.